terraform/vault-setup.sh

```
@@ -0,0 +1,250 @@
+EKS_CLUSTER_NAME=$(terraform output -raw eks_cluster_name)
+EKS_HOST=$(terraform output -raw eks_cluster_endpoint)
+EKS_CA_CERT=$(terraform output -raw eks_cluster_certificate_authority_data | base64 -d)
+
+# Configure Kubernetes authentication with proper service account
+kubectl create namespace vault-system || true
+kubectl create serviceaccount vault-auth -n vault-system || true
+kubectl create clusterrolebinding vault-auth-delegator --clusterrole=system:auth-delegator --serviceaccount=vault-system:vault-auth || true
+
+# Get service account token
+TOKEN_REVIEW_JWT=$(kubectl get secret \
+    $(kubectl get serviceaccount vault-auth -n vault-system -o jsonpath='{.secrets[0].name}') \
```

Shellcheck (0.10.0)

[warning] Quote this to prevent word splitting.

(SC2046)